integration_test/agents/agents.go

```diff
@@ -786,23 +825,43 @@ func SetupOpsAgent(ctx context.Context, logger *log.Logger, vm *gce.VM, config s
 	return SetupOpsAgentFrom(ctx, logger, vm, config, LocationFromEnvVars())
 }
 
-// restartOpsAgent restarts the Ops Agent and waits for it to become available.
-func restartOpsAgent(ctx context.Context, logger *log.Logger, vm *gce.VM) error {
+// RestartOpsAgent restarts the Ops Agent and waits for it to become available.
+func RestartOpsAgent(ctx context.Context, logger *log.Logger, vm *gce.VM) error {
 	if _, err := gce.RunRemotely(ctx, logger, vm, getRestartOpsAgentCmd(vm.ImageSpec)); err != nil {
-		return fmt.Errorf("restartOpsAgent() failed to restart ops agent: %v", err)
+		return fmt.Errorf("RestartOpsAgent() failed to restart ops agent: %v", err)
 	}
 	// Give agents time to shut down. Fluent-Bit's default shutdown grace period
 	// is 5 seconds, so we should probably give it at least that long.
 	time.Sleep(10 * time.Second)
 	return nil
 }
 
+func getStartOpsAgentPluginCmd(imageSpec string, port string) string {
+	if gce.IsWindows(imageSpec) {
+		return ""
+	}
+	return fmt.Sprintf("sudo nohup ~/plugin --address=localhost:%s --errorlogfile=errorlog.txt --protocol=tcp 1>/dev/null 2>/dev/null &", port)
+}
+
+// StartOpsAgentPlugin starts the Ops Agent Plugin gRPC server on the testing VM in the background.
+func StartOpsAgentPlugin(ctx context.Context, logger *log.Logger, vm *gce.VM, port string) error {
+	if _, err := gce.RunRemotely(ctx, logger, vm, getStartOpsAgentPluginCmd(vm.ImageSpec, port)); err != nil {
+		return fmt.Errorf("StartOpsAgentPlugin() failed to start the ops agent plugin: %v", err)
+	}
+	// Give agents time to shut down. Fluent-Bit's default shutdown grace period
```

Comment:
I don't understand this comment. Is this function starting the  agent? why do we wait for Fluent-bit shutdown

Reply:
you are right, this function only starts the agent, removed the 10 seconds wait.